File: api/src/main/java/com/glancebar/example/api/service/BookService.java
package com.glancebar.example.api.service;

import com.glancebar.commons.CommonHandling;
import com.glancebar.example.api.dto.BookDTO;

import java.util.List;

/**
 * 简单示例
 *
 * @author YISHEN CAI
 */
public interface BookService {

    /**
     * 关键字模糊搜索
     *
     * @param kw
     *            搜索关键字
     * 
     * @return book列表
     */
    List<BookDTO> searchByMatch(String kw);

    BookDTO addBook(BookDTO bookDTO);
}

File: api/src/main/kotlin/com/glancebar/example/api/service/BookService.kt
package com.glancebar.example.api.service

import com.glancebar.example.api.dto.BookDTO
import javax.validation.GroupSequence


/**
 *
 * @author YISHEN CAI
 */
interface BookService {
    fun searchByMatch(kw: String): List<BookDTO>

    annotation class Save {
    }
    fun addBook(bookDTO: BookDTO): Boolean
}